### PR TITLE
Add support for storing matrices and quaternions in AttributeArrays

### PR DESCRIPTION
**Pull request by danrbailey**

### CODE REVIEW

**Comment by danrbailey** [comment]
@khang @Idclip @jbrd

**Comment by Idclip** [comment]
Looks good to me. Approved.

File: openvdb_points/unittest/TestAttributeArray.cc [hunk 1]
```diff
@@ -1675,6 +1679,86 @@ TestAttributeArray::testDelayedLoad()
...
+
+void
+TestAttributeArray::testQuaternions()
+{
+    using namespace openvdb;
+
+    typedef TypedAttributeArray<math::Quat<float> >         AttributeQF;
+    typedef TypedAttributeArray<QuatR>                      AttributeQD;
+
+    AttributeQF::registerType();
```
Comment (jbrd) [inline openvdb_points/unittest/TestAttributeArray.cc]: Should we be adding these type registrations to openvdb::points::initialize?

File: openvdb_points/unittest/TestAttributeArray.cc [hunk 2]
```diff
@@ -1675,6 +1679,86 @@ TestAttributeArray::testDelayedLoad()
...
+
+
+void
+TestAttributeArray::testMatrices()
+{
+    using namespace openvdb;
+
+    typedef TypedAttributeArray<Mat4d>      AttributeM;
+
+    AttributeM::registerType();
```
Comment (jbrd) [inline openvdb_points/unittest/TestAttributeArray.cc]: Same for this one...

**Comment by jbrd** [comment]
I've taken a look through this now - just a comment about the type registrations... Other than that, looks good.

File: openvdb_points/unittest/TestAttributeArray.cc (hunk 1 above)
Comment (danrbailey) [inline openvdb_points/unittest/TestAttributeArray.cc]: Maybe, I'm trying not go too crazy with the type registrations, but I guess we should add them though.

**Comment by khang** [comment]
Looks good. Thanks Dan!

**Pull request merged**